File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        left, right = 0, 1
        while right < len(prices):
            delta = prices[right] - prices[left]
            if delta > 0:
                max_profit = max(delta, max_profit)
                right += 1
            else:
                left = right
                right += 1
        
        return max_profit

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        let md = 0;
        if (!root) return md;

        const _helper = (root, count = 1) => {
            if (!root)  return count - 1;
            return 1 + Math.max(_helper(root.left), _helper(root.right));
        };
        return _helper(root);
    }
}
